### PR TITLE
Fetch and display data mission

### PR DESCRIPTION
**Pull request by ulises2607**
- Add the fetchMission function to get the API data.
- Add extra reducers to manage the pending, fulfilled, and rejected status.
- Add props in the MissionCard component
- Add useSelector and useEffect functions in the MissionList component to dispatch and get the API data obtaining the relevant data to create all the mission cards.

### CODE REVIEW

File: src/redux/missions/missionsSlice.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      .addCase(fetchMission.rejected, (state, action) => {
+        state.loading = false;
+        state.error = action.error.message;
+      });
+  },
+});
+
+export const { addMission } = missionsSlice.actions;
+
+export default missionsSlice;
```
Comment (thasipablo) [inline src/redux/missions/missionsSlice.js:44]: As you have more than one export in the file, you can add `export` directly before `const` on line 17 and delete this line. This improves the code readability and matches well with JavaScript best practices.

The code will be like:
```
export const missionsSlice = createSlice({
  name: 'mission',
  initialState,
  reducers: {
    addMission: (state, action) => {
      console.log(action.payload);
    },
  },
```

[It's not mandatory but just a good practice]

File: src/components/MissionList.jsx [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+        <th>Status</th>
+        <th>   </th>
+      </tr>
+      {missionItems.map((item) => (
+        <MissionCard
+          key={item.mission_id}
+          id={item.mission_id}
+          missionName={item.mission_name}
+          missionDescription={item.description}
+        />
```
Comment (thasipablo) [inline src/components/MissionList.jsx:30]: - [ ] Here you just pass a single object(mission) as a prop:
 ``` {missionItems.map((mission) => (
        <MissionCard
          key={mission.mission_id}
          mission={mission}
        /> ```

This way your code will be clean and very readable.

**Review by thasipablo — APPROVED** [review]
Hi @ulises2607 ,
I want through your code. You are doing a good job.
I left some suggestions for changes. They are not mandatory but they will improve the readability of the code.

Happy coding. Cheers :+1:

File: src/components/MissionList.jsx (hunk 2 above)
Comment (ulises2607) [reply src/components/MissionList.jsx:30]: Thank you Pablo! This is very helpful to me and I will do this 💪

**Pull request merged**